backend/Warehouse/src/main/java/dk/sdu/controllab/warehouse/service/WarehouseServiceImpl.java

```
@@ -11,62 +12,98 @@
 
     @Autowired
     private EmulatorServiceConfig soapClient;
-    // for the time being this method requires the content of the inventory to be in the format "Itemname,amount"
+    @Autowired
+    private WarehouseRepository warehouseRepository;
+    @Autowired
+    private ItemService itemService;
     @Override
     public GetInventoryResponse getInventory() {
         GetInventoryResponse getInventoryResponse = new GetInventoryResponse();
-        String responseBody = soapClient.emulatorService().getInventory();
+        List<WarehouseInventoryEntry> inventoryList = warehouseRepository.findAll();
         Gson gson = new Gson();
+        getInventoryResponse.setItems(inventoryList);
+        String responseBody = soapClient.emulatorService().getInventory();
         JsonObject jsonObject = gson.fromJson(responseBody, JsonObject.class);
-        JsonArray inventoryArray = jsonObject.getAsJsonArray("Inventory");
         int state = jsonObject.get("State").getAsInt();
-
-        List<WarehouseInventoryEntry> inventoryList = new ArrayList<>();
         getInventoryResponse.setState(WarehouseState.values()[state]);
 
-        for (JsonElement element : inventoryArray) {
-            JsonObject item = element.getAsJsonObject();
-            String content = item.get("Content").getAsString();
-            int ID = item.get("Id").getAsInt();
-            String[] contentArray = content.split(",");
-            Item item1 = new Item();
-            item1.setName(contentArray[0]);
-            item1.setPredefinedTray((item.get("Id").getAsInt()));
-            inventoryList.add(new WarehouseInventoryEntry(1,item1,Integer.parseInt(contentArray[1]),ID));
-        }
-        getInventoryResponse.setItems(inventoryList);
-
         return getInventoryResponse;
 
     }
 
     @Override
-    public WarehouseInventoryEntry requestItem(Item item, int amount) {
+    public WarehouseInventoryEntry requestItem(UUID id) {
         return null;
```

Maybe throw an exception here, to indicate not implemented